### ISSUE TITLE
Load dependencies dynamically

### ISSUE THREAD

**Issue by grav** [grav/aws-lumo-cljs-runtime#9]
### Current limitations
The current runtime does not support loading dependencies in any way. But it is possible to load e.g. Promesa by adding the promesa jar to the function zip, and updating the `bootstrap` file to something like :

```
$bin_path/lumo --classpath .:/opt:promesa-2.0.1.jar ...
```

That said, it would be nice to be able to dynamically load dependencies (like NodeJS enables you to load whatever's in `node_modules`) without having to hack the runtime every time.

### Hence, a suggestion

Lumo can load dependencies dynamically using the `lumo.classpath/add!` api. Maybe we could make a convention where a `deps.cljs` file is loaded on bootup, which in turn loads all necessary dependencies using the aforementioned api? Like so:

`bootstrap` supplied in the runtime:
```
...
$bin_path/lumo --classpath .:/opt --init deps.cljs  ...
```

`deps.cljs` (or something similar) supplied in the lambda function zip:
```
(require '[lumo.classpath :as cp])
(cp/add! "promesa-2.0.1.jar")
(cp/add! "....jar") 
...
```

... or the user could specify the dependencies in a text file, which the runtime would parse and load in a `doseq` ... lots of possibilities for a convention.

**Comment by grav** [comment]
Btw, this should also support loading deps from another layer, since these are (as far as I can parse the documentation from AWS) just stuffed into `/opt`

https://docs.aws.amazon.com/lambda/latest/dg/configuration-layers.html

So, adding a layer with just `promesa-2.0.1.jar` would just mean appending`(cp/add! "/opt/promesa-2.0.1.jar")` to `deps.cljs`

**Comment by grav** [comment]
Edit: maybe we should just have a couple of magic directories, like `/opt/lib` and `lib`, and the runtime would just auto-load every jar in there (and of course log out what's been loaded).

**Comment by grav** [comment]
Got something working here (`bootstrap`)
```
#!/bin/bash
set -euo pipefail

function join { local IFS="$1"; shift; echo "$*"; }

bin_dir=${BIN_DIR:-/opt}
app_parts=(${_HANDLER//\// })
libs=$(join : $(find "$bin_dir/lib" lib -name '*.jar' 2>/dev/null))
classpath=".:${bin_dir}:${libs:-.}"

"$bin_dir/lumo" --classpath "$classpath" -e "(require (quote ${app_parts[0]}))" -m runtime
```

So the bootstrap script looks in `/opt/lib` and `lib`. Publishing a layer with just eg `lib/promesa-2.0.1.jar` will put it in `/opt/lib`, so a lambda function with a reference that layer and the runtime layer will be able to find the jar.

**Comment by grav** [comment]
I'll do a PR with these suggestions, once #3 is merged in.

**Comment by arichiardi** [comment]
I like the latter approach, the one with the working example!

This also opens the door to a small configuration file, the first entry of which can be that folder you have hardcoded for now. The others could be all the parameters that the `lumo` binary accepts.

Maybe a bit of overkill but some of the options (like `--static-fns`) might need some tweaking. Dunno I will think about it a bit more.

**Comment by grav** [comment]
Closed in #12
